File: apps/admin_mobile/lib/domain/repositories/super_admin_repository.dart
import '../entities/tenant.dart';
import '../entities/user.dart';
import '../../shared/types/paginated_result.dart';

abstract class SuperAdminRepository {
  Future<Map<String, dynamic>> getStats();
  Future<PaginatedResult<Tenant>> getTenants({int? page});
  Future<Tenant> suspendTenant(int id);
  Future<Tenant> activateTenant(int id);
  Future<PaginatedResult<User>> getUsers({int? page});
}

File: apps/admin-v2/src/domain/repositories/super-admin.repository.ts
import type { Tenant } from '../entities/tenant';
import type { User } from '../entities/user';
import type { PaginatedResult } from '../../shared/types/api';

export interface SuperAdminStats {
  totalTenants: number;
  activeTenants: number;
  totalUsers: number;
  totalReservations: number;
  totalServices: number;
}

export interface SuperAdminRepository {
  getStats(): Promise<SuperAdminStats>;
  getTenants(page?: number): Promise<PaginatedResult<Tenant>>;
  suspendTenant(id: string): Promise<Tenant>;
  activateTenant(id: string): Promise<Tenant>;
  getUsers(page?: number): Promise<PaginatedResult<User>>;
}
